src/planscape/plan/views.py

```diff
@@ -291,12 +289,29 @@ def update_project(request: HttpRequest) -> HttpResponse:
 
         _validate_constraint_values(
             max_budget, max_treatment_area_ratio, max_road_distance, max_slope)
-
-        project.priorities.clear()
-        _set_project_parameters(max_budget, max_treatment_area_ratio,
-                                max_road_distance, max_slope, priorities, project)
-        project.save()
-        return HttpResponse(str(project.pk))
+        
+        if request.method == "PUT":
+            project.priorities.clear()
+            _set_project_parameters(max_budget, max_treatment_area_ratio,
+                                    max_road_distance, max_slope, project)
+            _set_priorities(priorities, project)
+            project.save()
+            return HttpResponse(str(project.pk))
+        elif request.method == "PATCH":
+            del body['id']
+            body.pop('priorities', None)
+            s = ProjectSerializer(project, data=body, partial=True)
+            s.is_valid()
```

Comment:
I think this should be "s.is_valid(raise_exception=True)"

Reply:
(and please add a test for an invalid body dict)

Reply:
Done